libs/ui/design-system/src/lib/input.tsx

```diff
@@ -1,28 +1,62 @@
-import type * as React from 'react'
-
-import { Input as InputPrimitive } from '@nmit-coursition/ui/primitives/input'
+import * as InputPrimitive from '@nmit-coursition/ui/primitives/input'
 import { Label } from '@nmit-coursition/ui/primitives/label'
 import { cn } from '@nmit-coursition/ui/utils'
+import { type VariantProps, cva } from 'class-variance-authority'
+
+type OverrideProps<T, V> = V & Omit<T, keyof V>
 
-type InputType = React.ComponentProps<'input'>['type']
+const inputVariants = cva('', {
+  variants: {
+    /** todo: add orientation to the variants with mixins. */
```

Comment:
**suggestion:** Now that we don't have any variants we might use `orientation` straight away.

**suggestion:** I'd like to introduce one more called `grow` which will signalize if it should stay minimal or grow to the maximum size of its container. I have example in branch `add-design-system-button`.

Reply:
orientations are not useful in this case